mathics/core/symbols.py

```diff
@@ -490,8 +430,21 @@ def evaluate(self, evaluation):
                 return result.evaluate(evaluation)
         return self
 
-    def is_true(self) -> bool:
-        return self is SymbolTrue
+    def get_head(self) -> "Symbol":
+        return Symbol("Symbol")
+
+    def get_head_name(self):
+        return "System`Symbol"
+
+    def has_changed(self, definitions):
+        """
+        Used in Expression.evaluate() to determine if we need to reevaluation
+        an expression.
```

Comment:
Maybe false here too?

Reply:
Symbol is probably one of the very few places where we might have to worry about. You can update a symbol's definition during evaluation. 

But the thing that is a mystery (and therefore may be bad coding) is why this definition kind of check isn't done inside this method. I am guessing that in those cases where things change it is noticed in the caller. 

So has_changed may really mean "could possibly have changed".